mongo/client_test.go

```diff
@@ -344,4 +344,49 @@ func TestClient(t *testing.T) {
 			})
 		}
 	})
+	t.Run("serverAPI version", func(t *testing.T) {
+		t.Run("success", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion(options.ServerAPIVersion1)
+			client, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
+			assert.Nil(t, err, "unexpected error from NewClient: %v", err)
+			assert.Equal(t, ServerAPIOptions, client.serverAPI,
+				"mismatch in serverAPI; expected %v, got %v", ServerAPIOptions, client.serverAPI)
+		})
+		t.Run("success with more options", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion(options.ServerAPIVersion1).
+				SetStrict(true).SetDeprecationErrors(true)
+			client, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
+			assert.Nil(t, err, "unexpected error from NewClient: %v", err)
+			assert.Equal(t, ServerAPIOptions, client.serverAPI,
+				"mismatch in serverAPI; expected %v, got %v", ServerAPIOptions, client.serverAPI)
+		})
+		t.Run("failure with missing version", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetStrict(true).SetDeprecationErrors(true)
+			_, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
+			assert.NotNil(t, err, "expected error from NewClient, got nil")
+			errmsg := "api version \"\" not supported by driver"
+			assert.Equal(t, errmsg, err.Error(), "expected error %v, got %v", errmsg, err.Error())
+		})
+		t.Run("failure with unsupported version", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion("badVersion")
+			_, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
+			assert.NotNil(t, err, "expected error from NewClient, got nil")
+			errmsg := "api version \"badVersion\" not supported by driver"
+			assert.Equal(t, errmsg, err.Error(), "expected error %v, got %v", errmsg, err.Error())
+		})
+		t.Run("cannot modify options after client creation", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion(options.ServerAPIVersion1).
```

Comment:
The same options are declared twice in this test: once here and once for `expectedServerAPIOptions`. Can we refactor to use an anonymous function:

```
getServerAPIOptions := func() *options.ServerAPIOptions {
    return options.ServerAPI()...
}
```

and then use this wherever you need to get the base set of options?

Reply:
That sounds good, as I do get the base option quite a few times. Defined `getServerAPIOptions()` within the scope of these tests.